### PR TITLE
ConvertPrecision transformation: handle Assign, ReadValue and Variable

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
MicrosoftTeams-image
MicrosoftTeams-image (1)

+ leftovers from https://github.com/openvinotoolkit/openvino/pull/20404

### Tickets:
 - *CVS-125690*

### CODE REVIEW

File: src/plugins/intel_gpu/thirdparty/onednn_gpu
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/thirdparty/onednn_gpu:1]: please, revert

File: src/bindings/python/tests/test_runtime/test_infer_request.py [hunk 1]
```diff
@@ -673,9 +672,9 @@ def test_infer_queue_get_idle_handle(device):
     reason=f"Can't run test on device {os.environ.get('TEST_DEVICE', 'CPU')}, "
     "Memory layers fully supported only on CPU",
 )
-def test_query_state_write_buffer(device, input_shape, data_type, mode):
+def test_query_state_write_buffer(input_shape, data_type, mode):
     core = Core()
-
+    device = "CPU"
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/bindings/python/tests/test_runtime/test_infer_request.py]: debug code, reverted

File: src/plugins/intel_gpu/thirdparty/onednn_gpu
Comment (itikhono) [reply src/plugins/intel_gpu/thirdparty/onednn_gpu:1]: thanks! fixed

**Comment by wkobielx** [comment]
Running macos arm64 build+tests manually here: 
https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/ie/job/build-linux-macos_arm64/5928/

**Comment by itikhono** [comment]
@mateusztabaka @evkotov could you review this PR?

File: src/common/transformations/src/transformations/convert_precision.cpp [hunk 2]
```diff
@@ -557,6 +579,38 @@ bool fuse_type_to_parameter(const std::shared_ptr<ov::Node>& node,
...
 
+bool wrap_into_original_type(const std::shared_ptr<ov::Node>& node, const precisions_map& precisions) {
+    auto it = precisions.find(node->get_output_element_type(0));
+    if (it == precisions.end())
+        return false;
+
+    const auto& to = it->second;
+    const auto& from = it->first;
+
+    auto convert_before = std::make_shared<opset4::Convert>(node->input_value(0), from);
```
Comment (evkotov) [inline src/common/transformations/src/transformations/convert_precision.cpp:590]: Is this ok to use opset4 here? Should we use something about "ov::op::v0::Convert" instead?

File: src/common/transformations/src/transformations/convert_precision.cpp [hunk 3]
```diff
@@ -557,6 +579,38 @@ bool fuse_type_to_parameter(const std::shared_ptr<ov::Node>& node,
...
+    node->input(0).replace_source_output(convert_before);
+    auto consumers = node->output(0).get_target_inputs();
+    auto convert_after = std::make_shared<opset4::Convert>(node, to);
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/src/transformations/convert_precision.cpp:593]: minor.
I will update opset in the next PR

File: src/common/transformations/src/transformations/convert_precision.cpp (hunk 2 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/convert_precision.cpp:590]: minor.
I will update opset/version in the next PR

File: tools/mo/openvino/tools/mo/ops/read_value.py [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 # SPDX-License-Identifier: Apache-2.0
 import numpy as np
+from openvino.runtime import PartialShape
 
-from openvino.tools.mo.ops.op import Op
-from openvino.tools.mo.middle.passes.convert_data_type import np_data_type_to_destination_type
 from openvino.tools.mo.front.common.partial_infer.utils import unmask_shape
 from openvino.tools.mo.graph.graph import Graph, Node
-from openvino.runtime import PartialShape
+from openvino.tools.mo.middle.passes.convert_data_type import np_data_type_to_destination_type
```
[the remark this answers is not in the document]
Comment (itikhono) [reply tools/mo/openvino/tools/mo/ops/read_value.py:8]: this is a leftover from the previous PR

**Pull request merged**